### PR TITLE
Recursive clone in Dockerfiles, use build target for tofnd

### PR DESCRIPTION
**Pull request by agouin**
PR into @danbryan 's working branch for adding tofnd. Simplifies build to use heighliner built-in clone and build process rather than manually using pre-build. Fixes build for arm64.

### CODE REVIEW

File: dockerfile/rust/Dockerfile [hunk 1]
```diff
@@ -45,7 +45,7 @@ ARG GITHUB_REPO
 ARG VERSION
 ARG BUILD_TIMESTAMP
 
-RUN git clone -b ${VERSION} --single-branch https://${REPO_HOST}/${GITHUB_ORGANIZATION}/${GITHUB_REPO}.git
+RUN git clone -b ${VERSION} --single-branch https://${REPO_HOST}/${GITHUB_ORGANIZATION}/${GITHUB_REPO}.git --recursive
```
Comment (danbryan) [inline dockerfile/rust/Dockerfile:48]: this will make all rust apps to do a recursive pull.  Is that what we want?

File: chains.yaml [hunk 2]
```diff
@@ -59,15 +59,11 @@
   github-organization: axelarnetwork
   github-repo: tofnd
   language: rust
-  pre-build: |
-    apt install -y zlib1g-dev:${TARGETARCH}
-    TOOLCHAIN=$(cat rust-toolchain.toml | grep channel | awk '{print $3}' | tr -d '"')
-    rustup component add rust-src --toolchain ${TOOLCHAIN}-$(uname -m)-unknown-linux-gnu
-    git clone https://github.com/axelarnetwork/tofnd.git --recursive
```
Comment (danbryan) [inline chains.yaml:66]: why was this removed?  This builds and runs for me

**Review by danbryan — CHANGES_REQUESTED** [review]
added a few comments we should chat about before changing.

**Comment by DavidNix** [comment]
Also, a sentence or two in the PR description saying *why* the changes are needed. It's not obvious to me what recursive clone is solving.

File: dockerfile/rust/Dockerfile (hunk 1 above)
Comment (agouin) [reply dockerfile/rust/Dockerfile:48]: yes. If a repo does not have submodules, this is a noop. We want to make sure all build deps are cloned. These steps are done in the build repo, so no storage concerns in the final image.

File: chains.yaml (hunk 2 above)
Comment (agouin) [reply chains.yaml:66]: We can use the build-target for this, don't need to do this manually

**Comment by agouin** [comment]
> Also, a sentence or two in the PR description saying why the changes are needed. It's not obvious to me what recursive clone is solving.

For sure. This was a quick fix done over the weekend that @danbryan and I had chatted through, but will add a description.
